api/src/main/java/org/pmiops/workbench/api/DiskController.java

```diff
@@ -0,0 +1,121 @@
+package org.pmiops.workbench.api;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.logging.Logger;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.leonardo.model.LeonardoDiskStatus;
+import org.pmiops.workbench.leonardo.model.LeonardoGetPersistentDiskResponse;
+import org.pmiops.workbench.leonardo.model.LeonardoListPersistentDiskResponse;
+import org.pmiops.workbench.model.Disk;
+import org.pmiops.workbench.model.EmptyResponse;
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.notebooks.LeonardoNotebooksClient;
+import org.pmiops.workbench.utils.mappers.LeonardoMapper;
+import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DiskController implements DiskApiDelegate {
+  private static final Logger log = Logger.getLogger(DiskController.class.getName());
+  private final LeonardoNotebooksClient leonardoNotebooksClient;
+  private final WorkspaceDao workspaceDao;
+  private final WorkspaceAuthService workspaceAuthService;
+  private final LeonardoMapper leonardoMapper;
+
+  @Autowired
+  public DiskController(
+      LeonardoNotebooksClient leonardoNotebooksClient,
+      WorkspaceAuthService workspaceAuthService,
+      WorkspaceDao workspaceDao,
+      LeonardoMapper leonardoMapper) {
+    this.leonardoNotebooksClient = leonardoNotebooksClient;
+    this.workspaceAuthService = workspaceAuthService;
+    this.workspaceDao = workspaceDao;
+    this.leonardoMapper = leonardoMapper;
+  }
+
+  @Override
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
+    String firecloudWorkspaceName = dbWorkspace.getFirecloudName();
+    String googleProject = dbWorkspace.getGoogleProject();
+    workspaceAuthService.enforceWorkspaceAccessLevel(
```

Comment:
sorry, that part I am not sure. need @calbach review: I am not sure when/how to use `enforceWorkspaceAccessLevel` for all leo calls. Should just workspace writer can view the runtime? If we already user user credential, does Leo/SAM already perform necessary authZ check?

Reply:
> How you guarantee the last created disk is the one we want? 10 users are using the same workspace, and create PDs at the same time. What would happen then?

- The workspace is exclusively used by one user. So it won't happen that multiple users creating multiple PDs at the same time
- But multiple PDs would be created by the same user when
  - Directly calling the backend API to create PD
  - Trigger creating on multiple devices/browsers at the same time before the cache refreshed

Unfortuantely, we can't perfectly solve the second case for now. Because when we want to keep the uniqueness of PD while creating a new one with a unique UUID in a "copy-on-write" way. We need an atomic/transactional operation to guarantee the "compare-and-set/checking-then-creating" operation. But Leo doesn't provide it. 

But we could still use some alternative way to decrease the risk of creating multiple PDs somehow.
- Call getPD API to check the PD existence before Creating a new one. As @calbach mentioned in the UUID PR.
- Use cron job to monitor multiple PDs existence. (previously discussed with CH )